gateway/server.go

```diff
@@ -179,6 +180,9 @@ type Gateway struct {
 	// RedisController keeps track of redis connection and singleton
 	RedisController *storage.RedisController
 	hostDetails     hostDetails
+
+	healthCheckInfo atomic.Value
+	healthCheckLock sync.Mutex
```

Comment:
if `healthCheckInfo` is atomic, why do we need a mutex to guard it?

Reply:
good point @furkansenharputlu , probably old code. Removed the mutex 👍